v2/tools/generator/internal/astmodel/property_reference_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestPropertyReference_IsEmpty_ReturnsExpectedResult(t *testing.T) {
+	pkg := makeTestLocalPackageReference("Demo", "v1")
+	declaringType := MakeTypeName(pkg, "Person")
+
+	cases := []struct {
+		name          string
+		declaringType TypeName
+		property      PropertyName
+		expected      bool
+	}{
+		{"Property name present means not empty", EmptyTypeName, "property", false},
+		{"Declaring type present means not empty", declaringType, "", false},
+		{"Fully populated means not empty", declaringType, "property", false},
+		{"Totally empty not empty", EmptyTypeName, "", true},
+	}
+
+	t.Parallel()
+	for _, c := range cases {
+		c := c
+		t.Run(c.name, func(t *testing.T) {
+			t.Parallel()
+			g := NewGomegaWithT(t)
+			ref := MakePropertyReference(c.declaringType, c.property)
+			g.Expect(ref.IsEmpty()).To(Equal(c.expected))
+		})
+	}
+}
+
+func TestPropertyReference_String_ReturnsExpectedResult(t *testing.T) {
+	g := NewGomegaWithT(t)
```

Comment:
minor: Missing t.parallel here

Reply:
Ok so, I went on a whole journey regarding this (see my message in Teams). Feel free to ignore these comments if you like. I think @porges actually pointed out there's not a lot of win to t.Parallel() in fast tests anyway at some point in the past but I had forgotten.

Reply:
Fixed. Best to have them in place if this is our practice.

Reply:
I'm not actually 100% sure what our practice is, TBH. I'll ask @porges in our meeting today. 

I we want, I can turn on a linter that enforces this (I have it locally), but we _are_ missing it in a lot of places and it's not clear that it actually makes things faster. They seem to recommend it for "slow" tests, but not for "fast enough" tests.